pkg/utils/options/options.go

```diff
@@ -44,6 +44,7 @@ func MustParse() Options {
 	flag.StringVar(&opts.AWSNodeNameConvention, "aws-node-name-convention", env.WithDefaultString("AWS_NODE_NAME_CONVENTION", string(IPName)), "The node naming convention used by the AWS cloud provider. DEPRECATION WARNING: this field may be deprecated at any time")
 	flag.BoolVar(&opts.AWSENILimitedPodDensity, "aws-eni-limited-pod-density", env.WithDefaultBool("AWS_ENI_LIMITED_POD_DENSITY", true), "Indicates whether new nodes should use ENI-based pod density")
 	flag.StringVar(&opts.AWSDefaultInstanceProfile, "aws-default-instance-profile", env.WithDefaultString("AWS_DEFAULT_INSTANCE_PROFILE", ""), "The default instance profile to use when provisioning nodes in AWS")
+	flag.BoolVar(&opts.AWSEnablePodENI, "aws-enable-pod-eni", env.WithDefaultBool("AWS_ENABLE_POD_ENI", false), "If true then instances that support pod ENI will report a vpc.amazonaws.com/pod-eni resource")
```

Comment:
The pod-eni resources only appear from the vpc-cni if it's configured appropriately.  We need the customer to tell us if they configure it that way or not as I don't see a way to determine it ourselves.  If we always reported the resource, the node would never become "ready" from our POV as the ENI resource would never register.

Reply:
Technically possibly, but not easily.  The docs for enabling the pod ENI are [here](https://docs.aws.amazon.com/eks/latest/userguide/security-groups-for-pods.html). 

To do it, you set an environment variable on the aws-node DS.
```sh
kubectl set env daemonset aws-node -n kube-system ENABLE_POD_ENI=true
```

So to have a heterogeneous cluster, you'd need to edit the existing aws-node DS to not select all nodes and then create a duplicate DS which selected the other nodes and have only one of them configured with `ENABLE_POD_ENI=true`